presto-password-authenticators/src/main/java/io/prestosql/plugin/password/file/FileBasedAuthenticatorFactory.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.password.file;
+
+import com.google.inject.Injector;
+import com.google.inject.Scopes;
+import io.airlift.bootstrap.Bootstrap;
+import io.airlift.log.Logger;
+import io.airlift.units.Duration;
+import io.prestosql.spi.PrestoException;
+import io.prestosql.spi.security.PasswordAuthenticator;
+import io.prestosql.spi.security.PasswordAuthenticatorFactory;
+
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+
+import static com.google.common.base.Suppliers.memoizeWithExpiration;
+import static com.google.common.base.Throwables.throwIfUnchecked;
+import static io.airlift.configuration.ConfigBinder.configBinder;
+import static io.prestosql.spi.StandardErrorCode.CONFIGURATION_INVALID;
+import static java.lang.String.format;
+
+public class FileBasedAuthenticatorFactory
+        implements PasswordAuthenticatorFactory
+{
+    private static final Logger log = Logger.get(FileBasedAuthenticatorFactory.class);
+
+    private Supplier<PasswordAuthenticator> fileauthSupplier;
+
+    @Override
+    public String getName()
+    {
+        return "file";
+    }
+
+    @Override
+    public PasswordAuthenticator create(Map<String, String> config)
+    {
+        String configFileName = getName();
+        if (config.containsKey("file.refresh-period")) {
+            try {
+                Duration refreshPeriod = Duration.valueOf(config.get("file.refresh-period"));
+                if (refreshPeriod.toMillis() != 0) {
+                    fileauthSupplier = memoizeWithExpiration(
+                            () -> {
+                                log.debug("Refreshing FileBasedAuthFile from %s", configFileName);
+                                return createPasswordAuthenticator(config);
+                            },
+                            refreshPeriod.toMillis(),
+                            TimeUnit.MILLISECONDS);
+                    return (user, password) -> fileauthSupplier.get().createAuthenticatedPrincipal(user, password);
+                }
+            }
+            catch (IllegalArgumentException e) {
+                throw invalidRefreshPeriodException(config);
+            }
+        }
+
+        return createPasswordAuthenticator(config);
+    }
+
+    private PasswordAuthenticator createPasswordAuthenticator(Map<String, String> config)
+    {
+        try {
```

Comment:
We recently changed `Bootstrap` to not throw checked exceptions, so this try/catch block isn't needed anymore.

Reply:
@electrum , But try/catch is needed for `initialize` part of `Injector injector = app
                    .strictConfig()
                    .doNotInitializeLogging()
                    .setRequiredConfigurationProperties(config)
                    .initialize();`

Reply:
Done